sympy/integrals/prde.py

```diff
@@ -875,6 +875,31 @@ def parametric_log_deriv_heu(fa, fd, wa, wd, DE, c1=None):
 
     The argument w == Dtheta/theta
     """
+    # Note: ideally the code here should let n = 1 whenever possible, as the
+    # check in special_denom() only succeeds in that case.
+
+    # Special case when f and w are rational numbers (not in the book, but
+    # this fails this heuristic and comes up often enough to add here). In
+    # this case, we can set z = 0.
+    f = fa.as_expr()/fd.as_expr()
+    w = wa.as_expr()/wd.as_expr()
+    if f.is_Rational and w.is_Rational:
```

Comment:
What happens here if we have floats rather than rationals?

Reply:
I guess it would just fail this check in the heuristic, causing the integral to not be computed (as before). This particular function requires integer solutions (the resulting integer ends up being used as the power of a polynomial term). 

Floats in general are problematic in the Risch algorithm. I would say it's probably best if we just replaced all floats with rationals at the beginning of the algorithm.